smolvlm2 (#2690)
huggingface/blog: smolvlm2.md

---
title: "SmolVLM2 Family: Bringing Video Understanding to Every Device" 
thumbnail: /blog/assets/smolvlm2/banner.png
authors:
- user: orrzohar
  guest: true
  org: Stanford

> - user: mfarre

Author: set Orr as author

- user: andito
- user: merve
- user: pcuenq
- user: cyrilzakka
- user: xenova
---

# SmolVLM2 Family: Bringing Video Understanding to Every Device

## TL;DR: SmolVLM can now watch 📺 with even better visual understanding

SmolVLM2 represents a fundamental shift in how we think about video understanding - moving from massive models that require substantial computing resources to efficient models that can run anywhere. Our goal is simple: make video understanding accessible across all devices and use cases, from phones to servers.

We are releasing models in three sizes (2.2B, 500M and 256M), MLX ready from day zero.

To demonstrate our vision in small video models, we've built three practical applications that showcase the versatility of these models.

> ### iPhone Video Understanding

Reviewer: we should move the use cases right after section SmolVLM2 2.2B: Our New Star Player for Vision and Video, and put ToC right here imo for better readability

Reviewer: ## Table of Contents

- [SmolVLM2 Family: Bringing Video Understanding to Every Device](#smolvlm2-family-bringing-video-understanding-to-every-device)
  - [TL;DR: SmolVLM can now watch 📺 with even better visual understanding](#tldr-smolvlm-can-now-watch--with-even-better-visual-understanding)
  - [Table of Contents](#table-of-contents)
  - [Technical Details](#technical-details)
    - [SmolVLM2 2.2B: Our New Star Player for Vision and Video](#smolvlm2-22b-our-new-star-player-for-vision-and-video)
    - [Going Even Smaller: Meet the 500M and 256M Video Models](#going-even-smaller-meet-the-500m-and-256m-video-models)
    - [Suite of SmolVLM2 Apps]
        - [iPhone Video Understanding](#iphone-video-understanding)
        - [Video Highlight Generator](#video-highlight-generator)
        - [VLC media player integration](#vlc-media-player-integration)
  - [Using SmolVLM2 and Fine-tuning it with Transformers and MLX](#using-somlvlm2-and-fine-tuning-it-with-transformers-and-mlx)
    - [Transformers](#transformers)
      - [Video Inference](#video-inference)
      - [Multiple Image Inference](#multiple-image-inference)
      - [Interleaving Image, Text and Video](#interleaving-image-text-and-video)
    - [Inference with MLX](#inference-with-mlx)
    - [Fine-tuning SmolVLM2](#fine-tuning-smolvlm2)
  - [Read More](#read-more)

Reviewer: also no need to make collection a subsection, we can just mention in TLDR

Author: adjusted

<table style="border-collapse: collapse;">
<tr>
<td width="600" style="border: none;">
<center>
<iframe width="300" height="533" src="https://www.youtube.com/embed/hOj0U0cKapY" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</center>
</td>
<td valign="top" style="border: none;">

> We've created an iPhone app running SmolVLM2 completely locally. Using our 500M model, users can analyze and understand video content directly on their device - no cloud required. Interested in building iPhone video processing apps with AI models running locally? We're releasing it very soon - <a href="https://huggingface.co/datasets/HuggingFaceTB/smolvlm2-iphone-waitlist" target="_blank">fill this form to test and build with us!</a>

Author: great catch!

Want to try SmolVLM2 right away? Check out our [interactive chat interface](huggingface.co/spaces/HuggingFaceTB/SmolVLM2) where you can test visual and video understanding capabilities of SmolVLM2 2.2B through a simple, intuitive interface.
<br>

## Table of Contents

- [SmolVLM2 Family: Bringing Video Understanding to Every Device](#smolvlm2-family-bringing-video-understanding-to-every-device)
  - [TL;DR: SmolVLM can now watch 📺 with even better visual understanding](#tldr-smolvlm-can-now-watch--with-even-better-visual-understanding)
    - [iPhone Video Understanding](#iphone-video-understanding)
    - [Video Highlight Generator](#video-highlight-generator)
    - [VLC media player integration](#vlc-media-player-integration)
    - [SmolVLM2 Collection](#smolvlm2-collection)
  - [Table of Contents](#table-of-contents)
  - [Technical Details](#technical-details)
    - [SmolVLM2 2.2B: Our New Star Player for Vision and Video](#smolvlm2-22b-our-new-star-player-for-vision-and-video)
    - [Going Even Smaller: Meet the 500M and 256M Video Models](#going-even-smaller-meet-the-500m-and-256m-video-models)

>   - [Using SomlVLM2 and Fine-tuning it with Transformers and MLX](#using-somlvlm2-and-fine-tuning-it-with-transformers-and-mlx)

Reviewer: ```suggestion
  - [Using SmolVLM2 and Fine-tuning it with Transformers and MLX](#using-smolvlm2-and-fine-tuning-it-with-transformers-and-mlx)
```

Author: adjusted

    - [Fine-tuning SmolVLM2](#fine-tuning-smolvlm2)
  - [Read More](#read-more)

## Technical Details

We are introducing three new models with 256M, 500M and 2.2B parameters. The 2.2B model is the go-to choice for vision and video tasks, while the 500M and 256M models represent **the smallest video language models ever released**.

While they're small in size, they outperform any existing models per memory consumption. Looking at Video-MME (the go-to scientific benchmark in video), SmolVLM2 joins frontier model families on the 2B range and we lead the pack in the even smaller space.

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/smolvlm2-videomme2.png" width="50%" alt="SmolVLM2 Performance">

*Video-MME stands out as a comprehensive benchmark due to its extensive coverage across diverse video types, varying durations (11 seconds to 1 hour), multiple data modalities (including subtitles and audio), and high-quality expert annotations spanning 900 videos totaling 254 hours. Learn more [here](https://video-mme.github.io/home_page.html).*

### SmolVLM2 2.2B: Our New Star Player for Vision and Video

> The new 2.2B model got better at solving math problems with images, reading text in photos, understanding complex diagrams, and tackling scientific visual questions, we see this reflected in the model performance across different benchmarks:

Author: adjusted

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/smolvlm2-score-gains.png" width="50%" alt="SmolVLM2 Vision Score Gains">

> When it comes to video tasks, 2.2B is a good bang for the buck. Across the different scientific benchmarks where we evaluated it we want to highlight its performance on Video-MME where it outperforms all existing 2B models. 

Author: adjusted

> We were able to achieve a good balance on video/vision performance thanks to the data mixture learnings published in [Apollo: An Exploration of Video Understanding in Large Multimodal Models](https://apollo-lmms.github.io/)

Author: exactly :)

It’s so memory efficient, that you can run it even in a free Google Colab.

<details>
<summary>Python Code</summary>

```python
from transformers import AutoProcessor, AutoModelForImageTextToText

model_path = "HuggingFaceTB/SmolVLM2-2.2B-Instruct"
processor = AutoProcessor.from_pretrained(model_path)
model = AutoModelForImageTextToText.from_pretrained(
    model_path,
    torch_dtype=torch.bfloat16,
    _attn_implementation="flash_attention_2"
).to("cuda")

messages = [
    {
        "role": "user",
        "content": [
            {"type": "video", "path": "path_to_video.mp4"},
            {"type": "text", "text": "Describe this video in detail"}
        ]
    },
]

> inputs = processor.apply_chat_template(

Reviewer 2: Love the single-step pre-processing API!

    messages,
    add_generation_prompt=True,
    tokenize=True,
    return_dict=True,
    return_tensors="pt",
).to(model.device)

generated_ids = model.generate(**inputs, do_sample=False, max_new_tokens=64)
generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```
</details>

### Going Even Smaller: Meet the 500M and 256M Video Models

Nobody dared to release such small video models until today.

Our new [SmolVLM2-500M-Video-Instruct](https://huggingface.co/HuggingFaceTB/SmolVLM2-500M-Video-Instruct) model has very close video capabilities than the original SmolVLM 2.2B, but at a fraction of the size: we're getting the same video understanding capabilities with less than a quarter of the parameters.

> And then there's our little experiment, the SmolVLM2-256M-Video-Instruct. Think of it as our "what if" project - what if we could push the boundaries of small models even further? Taking inspiration from what [IBM achieved](https://ds4sd.github.io/docling/examples/pictures_description/) with our base [SmolVLM-256M-Instruct](https://huggingface.co/HuggingFaceTB/SmolVLM-256M-Instruct) a few weeks ago, we wanted to see how far we could go with video understanding. While it's more of an experimental release, we're hoping it'll inspire some creative applications and specialized fine-tuning projects.

Reviewer 2: Would it make sense to release a non-instruct version of the 256M variant then?

> ## Using SomlVLM2 and Fine-tuning it with Transformers and MLX

Reviewer: ```suggestion
## Using SmolVLM2 and Fine-tuning it with Transformers and MLX
```

Author: adjusted

We make SmolVLM2 available to use with transformers and MLX from day zero. In this section, you can find different inference alternatives and tutorials for video and multiple images.

### Transformers

> There are two ways to infer SmolVLM2 models, one is through a chat template and the other one gives you more control by passing in media through the processor.

Reviewer 2: ```suggestion
The easiest way to run inference with the SmolVLM2 models is through the conversational API – applying the chat template takes care of preparing all inputs automatically.
```

Removing the processor use because we are not showing it later.

You can load the model as follows.

```python

from transformers import AutoProcessor, AutoModelForImageTextToText

processor = AutoProcessor.from_pretrained(model_path)
model = AutoModelForImageTextToText.from_pretrained(
    model_path,
    torch_dtype=torch.bfloat16,

>     _attn_implementation="flash_attention_2"

Reviewer 2: same question

> ).to(DEVICE)

Reviewer 2: `DEVICE` is not defined

```

#### Video Inference

> You can pass videos through a chat template by passing in {“type”: “video”, “path”:{video_path}”. See below complete example. 

Reviewer 2: Is it possible to also use `url` and `video` for video URLs and video data, respectively, like we do for images?

> You can pass videos through a chat template by passing in {“type”: “video”, “path”:{video_path}”. See below complete example. 

Reviewer 2: ```suggestion
You can pass videos in your conversation using messages defined as `{"type": "video", "path": "{video_path}"`. See a complete example below. 

As we mentioned before, all you have to do is apply the chat template. Under the hood, this will take care of using the model's pre-processor to prepare the video and text inputs the way the model expects.
```

Reviewer 2: Maybe we should also mention the importance of the system prompt somewhere.

```python
import torch

messages = [
    {
        "role": "user",
        "content": [
            {"type": "video", "path": "path_to_video.mp4"},
            {"type": "text", "text": "Describe this video in detail"}
        ]
    },
]

inputs = processor.apply_chat_template(
    messages,
    add_generation_prompt=True,
    tokenize=True,
    return_dict=True,
    return_tensors="pt",
).to(model.device)

generated_ids = model.generate(**inputs, do_sample=False, max_new_tokens=64)
generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```

#### Multiple Image Inference

> You can infer multiple images through a chat template as well. 

Reviewer 2: ```suggestion
In addition to video, SmolVLM2 supports multi-image conversations. You can use the same API through the chat template.
```

```python
import torch

messages = [
    {
        "role": "user",
        "content": [
{"type": "text", "text": "What are the differences between these two images?"},
            {"type": "image", "path": "image_1.png"},
{"type": "image", "path": "image_2.png"}
            
        ]
    },
]

inputs = processor.apply_chat_template(
    messages,
    add_generation_prompt=True,
    tokenize=True,
    return_dict=True,
    return_tensors="pt",
).to(model.device)

generated_ids = model.generate(**inputs, do_sample=False, max_new_tokens=64)
generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```

#### Interleaving Image, Text and Video

> You can interleave image, video and text together by passing in `<image>` and `<video>` tokens inside text, cutting text through and inserting image lines in between.

Reviewer 2: Does this work? I thought that video and images were not supported at the same time.

Author: @merveenoyan did you successfully tried this? 
I would imagine is quite sensitive to the text that you put between the video and the image

Author: agree, removing

```python
import torch

messages = [
    {
        "role": "user",
        "content": [
{"type": "text", "text": "What is the similarity between this image <image>"},

            {"type": "image", "path": "image_1.png"},
{"type": "text", "text": "and this image <image>"},
{"type": "image", "path": "image_2.png"},
            
        ]
    },
]

> # infer like above

Reviewer 2: I'd show a complete snippet so people can copy/paste.

Author: added inference code

```
### Inference with MLX 

You can run SmolVLM2 on MLX with following one-liner: 
```bash

> python3 -m mlx_vlm.generate –model HuggingFaceTB/SmolVLM-256M-Instruct –max-tokens 400 –temp 0.0 -image {YOUR_IMAGE_PATH}

Reviewer 2: I'll come back to this with examples for video as well.

```

### Fine-tuning SmolVLM2

> 

Reviewer: I will add the video FT script to SmolLM repo today and we can link to my existing notebook for image FT to smol-vision

> 

Reviewer 2: Still missing cc @merveenoyan

> ## Read More

Author: adjusted

[SmolVLM2 - Collection with Models and Demos](https://huggingface.co/collections/HuggingFaceTB/smolvlm2-smallest-video-lm-ever-67ab6b5e84bf8aaa60cb17c7)

> [SmolVLM - small yet mighty Vision Language Model](https://huggingface.co/blog/smolvlm)

Reviewer 2: Maybe replace with a conclusion still. We can still link to the collection there, but not to the previous blog post imo (we can do it elsewhere in the post), since this would not be a more in-depth reading.

---
title: "SmolVLM2: Bringing Video Understanding to Every Device" 
thumbnail: /blog/assets/smolvlm2/banner.png
authors:
- user: orrzohar
  guest: true
  org: Stanford
- user: mfarre
- user: andito
- user: merve
- user: pcuenq
- user: cyrilzakka
- user: xenova
---

# SmolVLM2: Bringing Video Understanding to Every Device

## TL;DR: SmolVLM can now watch 📺 with even better visual understanding

SmolVLM2 represents a fundamental shift in how we think about video understanding - moving from massive models that require substantial computing resources to efficient models that can run anywhere. Our goal is simple: make video understanding accessible across all devices and use cases, from phones to servers.

We are releasing models in three sizes (2.2B, 500M and 256M), MLX ready (Python _and_ Swift APIs) from day zero.

> To demonstrate our vision in small video models, we've built three practical applications that showcase the versatility of these models.

Reviewer 3: ```suggestion
To demonstrate our vision for small video models, we've built three practical applications that showcase the versatility of these models.
```

### iPhone Video Understanding
<table style="border-collapse: collapse;">
<tr>
<td width="600" style="border: none;">
<center>
<iframe width="300" height="533" src="https://www.youtube.com/embed/G1yQlHTk_Ig" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</center>
</td>
<td valign="top" style="border: none;">

> We've created an iPhone app running SmolVLM2 completely locally. Using our 500M model, users can analyze and understand video content directly on their device - no cloud required. Interested in building iPhone video processing apps with AI models running locally? We're releasing it very soon - <a href="https://huggingface.co/datasets/HuggingFaceTB/smolvlm2-iphone-waitlist" target="_blank">fill this form to test and build with us!</a>

Reviewer 3: ```suggestion
We've created an iPhone app that runs SmolVLM2 completely locally. Using our 500M model, users can analyze and understand video content directly on their device — no cloud required. Interested in building iPhone video processing apps with AI models running locally? We're releasing it very soon — <a href="https://huggingface.co/datasets/HuggingFaceTB/smolvlm2-iphone-waitlist" target="_blank">fill in this form to test and build with us!</a>
```

<tr>
<td width="500" style="border: none;">
<center>
<iframe width="500" height="281" src="https://www.youtube.com/embed/ZT2oS8EqnKI" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</center>
</td>
<td valign="top" style="border: none;">
Available as a Hugging Face Space, this application takes long-form videos (1+ hours) and automatically extracts the most significant moments. We've tested it extensively with soccer matches and other lengthy events, making it a powerful tool for content summarization. <a href="https://huggingface.co/spaces/HuggingFaceTB/SmolVLM2-HighlightGenerator" target="_blank">Try it yourself in our demo space.</a>
</td>
</tr>
</table>

### VLC media player integration
<table style="border-collapse: collapse;">
<tr>
<td width="500" style="border: none;">
<center>
<iframe width="500" height="281" src="https://www.youtube.com/embed/NGHCFEW7DCg" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</center>
</td>
<td valign="top" style="border: none;">

> Working in collaboration with VLC media player, we're integrating SmolVLM2 to provide intelligent video segment descriptions and navigation. This integration allows users to search through video content semantically, jumping directly to relevant sections based on natural language descriptions. While this is work in progress, you can experiment with the current playlist builder prototype <a href="https://huggingface.co/spaces/HuggingFaceTB/SmolVLM2-XSPFGenerator" target="_blank">in this space.</a>

Reviewer 3: ```suggestion
Working in collaboration with VLC media player, we're integrating SmolVLM2 to provide intelligent video segment descriptions and navigation. This integration allows users to search through video content semantically, jumping directly to relevant sections based on natural language descriptions. While this is work in progress, you can experiment with the current playlist builder prototype with <a href="https://huggingface.co/spaces/HuggingFaceTB/SmolVLM2-XSPFGenerator" target="_blank">this space.</a>
```

    - [VLC media player integration](#vlc-media-player-integration)
    - [SmolVLM2 Collection](#smolvlm2-collection)
  - [Table of Contents](#table-of-contents)
  - [Technical Details](#technical-details)
    - [SmolVLM2 2.2B: Our New Star Player for Vision and Video](#smolvlm2-22b-our-new-star-player-for-vision-and-video)
    - [Going Even Smaller: Meet the 500M and 256M Video Models](#going-even-smaller-meet-the-500m-and-256m-video-models)
  - [Using SmolVLM2 and Fine-tuning it with Transformers and MLX](#using-smolvlm2-and-fine-tuning-it-with-transformers-and-mlx)
    - [Transformers](#transformers)
      - [Video Inference](#video-inference)
      - [Multiple Image Inference](#multiple-image-inference)
      - [Interleaving Image, Text and Video](#interleaving-image-text-and-video)
    - [Inference with MLX](#inference-with-mlx)
    - [Fine-tuning SmolVLM2](#fine-tuning-smolvlm2)
  - [Read More](#read-more)

## Technical Details

We are introducing three new models with 256M, 500M and 2.2B parameters. The 2.2B model is the go-to choice for vision and video tasks, while the 500M and 256M models represent **the smallest video language models ever released**.

> While they're small in size, they outperform any existing models per memory consumption. Looking at Video-MME (the go-to scientific benchmark in video), SmolVLM2 joins frontier model families on the 2B range and we lead the pack in the even smaller space.

Reviewer 3: Ah I see the footnote. Maybe add a link to it here.

generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```
</details>

### Going Even Smaller: Meet the 500M and 256M Video Models

Nobody dared to release such small video models until today.

Our new [SmolVLM2-500M-Video-Instruct](https://huggingface.co/HuggingFaceTB/SmolVLM2-500M-Video-Instruct) model has very close video capabilities than the original SmolVLM 2.2B, but at a fraction of the size: we're getting the same video understanding capabilities with less than a quarter of the parameters.

And then there's our little experiment, the SmolVLM2-256M-Video-Instruct. Think of it as our "what if" project - what if we could push the boundaries of small models even further? Taking inspiration from what [IBM achieved](https://ds4sd.github.io/docling/examples/pictures_description/) with our base [SmolVLM-256M-Instruct](https://huggingface.co/HuggingFaceTB/SmolVLM-256M-Instruct) a few weeks ago, we wanted to see how far we could go with video understanding. While it's more of an experimental release, we're hoping it'll inspire some creative applications and specialized fine-tuning projects.

> ## Using SmolVLM2 and Fine-tuning it with Transformers and MLX

Reviewer 2: ```suggestion
## Using SmolVLM2 with Transformers and MLX
```

maybe?

We make SmolVLM2 available to use with transformers and MLX from day zero. In this section, you can find different inference alternatives and tutorials for video and multiple images.

### Transformers

There are two ways to infer SmolVLM2 models, one is through a chat template and the other one gives you more control by passing in media through the processor.

You can load the model as follows.

```python

from transformers import AutoProcessor, AutoModelForImageTextToText

processor = AutoProcessor.from_pretrained(model_path)
model = AutoModelForImageTextToText.from_pretrained(
    model_path,
    torch_dtype=torch.bfloat16,
    _attn_implementation="flash_attention_2"
).to(DEVICE)
```

#### Video Inference

> You can pass videos through a chat template by passing in {“type”: “video”, “path”:{video_path}”. See below complete example. 

Reviewer 3: ```suggestion
You can pass videos through a chat template by passing in `{"type": "video", "path": {video_path}`. See below for a complete example. 
```

).to(model.device)

generated_ids = model.generate(**inputs, do_sample=False, max_new_tokens=64)
generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```

#### Interleaving Image, Text and Video

You can interleave image, video and text together by passing in `<image>` and `<video>` tokens inside text, cutting text through and inserting image lines in between.

```python
import torch

messages = [
    {
        "role": "user",
        "content": [
{"type": "text", "text": "What is the similarity between this image <image>"},

            {"type": "image", "path": "image_1.png"},
{"type": "text", "text": "and this image <image>"},
{"type": "image", "path": "image_2.png"},
            
        ]
    },
]

inputs = processor.apply_chat_template(
    messages,
    add_generation_prompt=True,
    tokenize=True,
    return_dict=True,
    return_tensors="pt",
).to(model.device)

generated_ids = model.generate(**inputs, do_sample=False, max_new_tokens=64)
generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```
### Inference with MLX 

> You can run SmolVLM2 on MLX with following one-liner: 
> ```bash
> python3 -m mlx_vlm.generate –model HuggingFaceTB/SmolVLM-256M-Instruct –max-tokens 400 –temp 0.0 -image {YOUR_IMAGE_PATH}

Reviewer 2: ```suggestion
To run SmolVLM2 with MLX on Apple Silicon devices using Python, you can use the excellent [mlx-vlm library](https://github.com/Blaizzy/mlx-vlm).

First, you need to install `mlx-vlm` from [this branch](https://github.com/Blaizzy/mlx-vlm/pull/208) using the following command:
 
 ```bash
pip install git+https://github.com/pcuenca/mlx-vlm.git@smolvlm
```
 
Then you can run inference on a single image using the following one-liner, which uses the [unquantized 500M version of SmolVLM2](mlx-community/SmolVLM2-500M-Video-Instruct-mlx):

```bash
python -m mlx_vlm.generate --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx --image https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/bee.jpg --prompt "Can you describe this image?"
```

The Swift language is also supported through the [mlx-swift-examples repo](https://github.com/ml-explore/mlx-swift-examples), which is what we used to build our iPhone app.

Until our PR is finalized and merged, you have to compile the project from [this fork](https://github.com/cyrilzakka/mlx-swift-examples), and then you can use the `llm-tool` CLI on your Mac like follows.

For image inference:

```bash
./mlx-run --debug llm-tool --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx --prompt "Can you describe this image?" --image https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/bee.jpg  --temperature 0.7 --top-p 0.9 --max-tokens 100
```

Video analysis is also supported, as well as providing a system prompt. We found system prompts to be particularly helpful for video understanding, to drive the model to the desired level of detail we are interested in. This is a video inference example:

```bash
./mlx-run --debug llm-tool --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx --system "Focus only on describing the key dramatic action or notable event occurring in this video segment. Skip general context or scene-setting details unless they are crucial to understanding the main action." --prompt "What is happening in this video?" --video /Users/pedro/Downloads/IMG_2855.mov  --temperature 0.7 --top-p 0.9 --max-tokens 100
```

If you integrate SmolVLM2 in your apps using MLX and Swift, we'd love to know about it! Please, feel free to drop us a note in the comments section below!

Reviewer 2: This suggestion got messed up with the code blocks, I'll push a commit.

    --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx \
    --prompt "Can you describe this image?" \
    --image https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/bee.jpg \
    --temperature 0.7 --top-p 0.9 --max-tokens 100
```

Video analysis is also supported, as well as providing a system prompt. We found system prompts to be particularly helpful for video understanding, to drive the model to the desired level of detail we are interested in. This is a video inference example:

```bash
./mlx-run --debug llm-tool \
    --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx \
    --system "Focus only on describing the key dramatic action or notable event occurring in this video segment. Skip general context or scene-setting details unless they are crucial to understanding the main action." \
    --prompt "What is happening in this video?" \
    --video /Users/pedro/Downloads/IMG_2855.mov \
    --temperature 0.7 --top-p 0.9 --max-tokens 100
```

If you integrate SmolVLM2 in your apps using MLX and Swift, we'd love to know about it! Please, feel free to drop us a note in the comments section below!

### Fine-tuning SmolVLM2

## Read More

> 

Reviewer: ```suggestion
We would like to thank Raushan Turganbay, Arthur Zucker and Pablo Montalvo Leroux for their contribution of the model to transformers.
```

Author: added

    --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx \
    --prompt "Can you describe this image?" \
    --image https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/bee.jpg \
    --temperature 0.7 --top-p 0.9 --max-tokens 100
```

Video analysis is also supported, as well as providing a system prompt. We found system prompts to be particularly helpful for video understanding, to drive the model to the desired level of detail we are interested in. This is a video inference example:

```bash
./mlx-run --debug llm-tool \
    --model mlx-community/SmolVLM2-500M-Video-Instruct-mlx \
    --system "Focus only on describing the key dramatic action or notable event occurring in this video segment. Skip general context or scene-setting details unless they are crucial to understanding the main action." \
    --prompt "What is happening in this video?" \
    --video /Users/pedro/Downloads/IMG_2855.mov \
    --temperature 0.7 --top-p 0.9 --max-tokens 100
```

If you integrate SmolVLM2 in your apps using MLX and Swift, we'd love to know about it! Please, feel free to drop us a note in the comments section below!

### Fine-tuning SmolVLM2

> 

Reviewer: ```suggestion
You can fine-tune SmolVLM2 on videos using transformers 🤗

We have fine-tuned 500M variant in Colab on video-caption pairs in [VideoFeedback](https://huggingface.co/datasets/TIGER-Lab/VideoFeedback) dataset for demonstration purposes. Since 500M variant is small, it's better to apply full fine-tuning instead of QLoRA or LoRA, meanwhile you can try to apply QLoRA on 2.2B variant. You can find the fine-tuning notebook [here](https://github.com/huggingface/smollm/blob/main/vision/finetuning/SmolVLM2_Video_FT.ipynb).
```
